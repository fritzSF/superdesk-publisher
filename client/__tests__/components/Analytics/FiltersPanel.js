import React from "react";
import FiltersPanel from "../../../components/Analytics/FiltersPanel";
import { render, fireEvent } from "@testing-library/react";

let api = () => {};

jest.mock("../../../components/UI/MultiSelect", () => () => {
  return <div></div>;
});

api.users = {};
api.users.query = function () {
  return new Promise((resolve, reject) => {
    resolve({
      _items: [{ is_author: true, display_name: "author" }],
      _links: {},
    });
  });
};

describe("Analytics/FiltersPanel", () => {
  const filters = {
    route: 1,
    published_after: "2019-07-04",
    published_before: "2019-07-04",
  };

  const routes = [
    {
      id: 1,
      name: "testroute",
    },
  ];

  it("renders correctly", () => {
    const { container } = render(
      <FiltersPanel
        toggle={jest.fn()}
        filters={filters}
        setFilters={jest.fn()}
        routes={routes}
        api={api}
      />
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it("fires toggle", async () => {
    const toggle = jest.fn();

    const { container } = render(
      <FiltersPanel
        toggle={toggle}
        filters={{}}
        setFilters={jest.fn()}
        routes={[]}
        api={api}
      />
    );

    const button = container.querySelector(".side-panel__close button");

    fireEvent.click(button);
    expect(toggle).toHaveBeenCalled();
  });

  it("updates and sets filters", () => {
    const setFilters = jest.fn();

    const { container, getByText } = render(
      <FiltersPanel
        toggle={jest.fn()}
        filters={filters}
        setFilters={setFilters}
        routes={routes}
        api={api}
      />
    );

    const filterButton = getByText("Run Report");
    fireEvent.click(filterButton);

    expect(setFilters).toHaveBeenCalled();
    expect(setFilters).toHaveBeenCalledWith(filters);
  });

  it("clears filters", () => {
    const setFilters = jest.fn();

    const { getByText } = render(
      <FiltersPanel
        toggle={jest.fn()}
        filters={filters}
        setFilters={setFilters}
        routes={routes}
        api={api}
      />
    );

    const button = getByText("Clear");
    fireEvent.click(button);

    expect(setFilters).toHaveBeenCalled();
    expect(setFilters).toHaveBeenCalledWith({});
  });
});
