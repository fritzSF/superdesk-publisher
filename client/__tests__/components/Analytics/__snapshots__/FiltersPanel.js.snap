// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Analytics/FiltersPanel renders correctly 1`] = `
<div
  class="sd-filters-panel sd-filters-panel--border-right"
>
  <div
    class="side-panel side-panel--transparent side-panel--shadow-right"
  >
    <div
      class="side-panel__header side-panel__header--border-b"
    >
      <span
        class="side-panel__close"
      >
        <button
          class="icn-btn"
          tabindex="0"
        >
          <i
            aria-hidden="true"
            aria-label="close-small"
            class="icon-close-small"
          />
        </button>
      </span>
      <h3
        class="side-panel__heading side-panel__heading--big"
      >
        Filter
      </h3>
    </div>
    <div
      class="side-panel__content"
    >
      <div
        class="side-panel__content-block"
      >
        <div
          class="form__row"
        >
          <div
            class="sd-line-input sd-line-input--no-margin"
          >
            <label
              class="sd-line-input__label"
            >
              Category
            </label>
            <div />
          </div>
        </div>
        <div
          class="form__row"
        >
          <div
            class="sd-line-input sd-line-input--no-margin"
          >
            <label
              class="sd-line-input__label"
            >
              Authors
            </label>
            <div
              class=" css-ewg31r-container"
            >
              <div
                class=" css-1bpltnu-control"
              >
                <div
                  class=" css-g1d714-ValueContainer"
                >
                  <div
                    class=" css-1wa3eu0-placeholder"
                  >
                    Select...
                  </div>
                  <div
                    class="css-b8ldur-Input"
                  >
                    <div
                      class=""
                      style="display: inline-block;"
                    >
                      <input
                        aria-autocomplete="list"
                        autocapitalize="none"
                        autocomplete="off"
                        autocorrect="off"
                        id="react-select-2-input"
                        spellcheck="false"
                        style="box-sizing: content-box; width: 2px; border: 0px; font-size: inherit; opacity: 1; outline: 0; padding: 0px;"
                        tabindex="0"
                        type="text"
                        value=""
                      />
                      <div
                        style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-size: inherit; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class=" css-1rkxiuv-IndicatorsContainer"
                >
                  <div
                    aria-hidden="true"
                    class=" css-at12u2-loadingIndicator"
                  >
                    <span
                      class="css-1yvy2vo-LoadingDot"
                    />
                    <span
                      class="css-zoievk-LoadingDot"
                    />
                    <span
                      class="css-x748d8-LoadingDot"
                    />
                  </div>
                  <span
                    class=" css-1xjgjl1-IndicatorSeparator"
                  />
                  <div
                    aria-hidden="true"
                    class=" css-tlfecz-indicatorContainer"
                  >
                    <svg
                      aria-hidden="true"
                      class="css-6q0nyr-Svg"
                      focusable="false"
                      height="20"
                      viewBox="0 0 20 20"
                      width="20"
                    >
                      <path
                        d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="form__row"
        >
          <div
            class="sd-line-input sd-line-input--no-margin sd-line-input--is-select"
          >
            <label
              class="sd-line-input__label"
            >
              Date filter
            </label>
            <select
              class="sd-line-input__select"
              name="route"
            >
              <option
                value="range"
              >
                Custom
              </option>
              <option
                value="lastWeek"
              >
                Last week
              </option>
              <option
                value="lastMonth"
              >
                Last month
              </option>
              <option
                value="thisWeek"
              >
                This week
              </option>
              <option
                value="thisMonth"
              >
                This month
              </option>
            </select>
          </div>
        </div>
        <div
          class="form__row form__row--flex"
        >
          <div
            class="form__row-item"
          >
            <div
              class="sd-line-input sd-line-input--no-margin"
            >
              <label
                class="sd-line-input__label"
              >
                From
              </label>
              <span
                class="p-calendar p-inputwrapper p-calendar-w-btn p-inputwrapper-filled"
                id="pr_id_1"
              >
                <input
                  autocomplete="off"
                  class="p-inputtext p-component"
                  inputmode="none"
                  type="text"
                />
                <button
                  class="p-button p-component p-datepicker-trigger p-button-icon-only"
                  tabindex="-1"
                  type="button"
                >
                  <span
                    class="icon-calendar p-c"
                  />
                  <span
                    class="p-button-label p-c"
                  >
                     
                  </span>
                </button>
              </span>
            </div>
          </div>
          <div
            class="form__row-item"
          >
            <div
              class="sd-line-input sd-line-input--no-margin"
            >
              <label
                class="sd-line-input__label"
              >
                To
              </label>
              <span
                class="p-calendar p-inputwrapper p-calendar-w-btn p-inputwrapper-filled"
                id="pr_id_2"
              >
                <input
                  autocomplete="off"
                  class="p-inputtext p-component"
                  inputmode="none"
                  type="text"
                />
                <button
                  class="p-button p-component p-datepicker-trigger p-button-icon-only"
                  tabindex="-1"
                  type="button"
                >
                  <span
                    class="icon-calendar p-c"
                  />
                  <span
                    class="p-button-label p-c"
                  >
                     
                  </span>
                </button>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="side-panel__footer side-panel__footer--button-box"
    >
      <div
        class="flex-grid flex-grid--boxed-small flex-grid--small-2"
      >
        <button
          aria-label=""
          class="btn btn--hollow"
          tabindex="0"
        >
          Clear
        </button>
        <button
          aria-label=""
          class="btn btn--primary"
          tabindex="0"
        >
          Run Report
        </button>
      </div>
      <div
        class="flex-grid flex-grid--boxed-small flex-grid--small-1"
        style="overflow: visible;"
      >
        <button
          aria-label=""
          class="btn btn--primary"
          tabindex="0"
        >
          Generate Report
        </button>
      </div>
    </div>
  </div>
</div>
`;
